Include/internal/pycore_frame.h

```diff
@@ -190,11 +190,11 @@ _PyFrame_FastToLocalsWithError(_PyInterpreterFrame *frame);
 void
 _PyFrame_LocalsToFast(_PyInterpreterFrame *frame, int clear);
 
-
 static inline bool
 _PyThreadState_HasStackSpace(PyThreadState *tstate, int size)
 {
```

Comment:
Looking good now!

Could you perhaps add an assert here that we expect either both `datastack_top` and `datastack_limit` to be NULL or none of them?  Since that's an invariant that this function relies on?

Reply:
Done